### PR TITLE
Update expected vs. actual type in checkTypeM

### PR DESCRIPTION
**Pull request by zliu41**
The expected and actual types were reversed. @michaelpj noticed it in https://github.com/input-output-hk/plutus/pull/4574#discussion_r863942380

### CODE REVIEW

File: plutus-core/plutus-core/src/PlutusCore/Error.hs [hunk 1]
```diff
@@ -86,7 +86,9 @@ data TypeError term uni fun ann
     = KindMismatch ann (Type TyName uni ()) (Kind ()) (Kind ())
     | TypeMismatch ann
         term
+        -- | Expected type
         (Type TyName uni ())
```
Comment (effectfully) [inline plutus-core/plutus-core/src/PlutusCore/Error.hs:90]: I think that [breaks](https://github.com/haskell/haddock/issues/1065) haddock?

**Review by effectfully — COMMENTED** [review]
I always get them wrong, it's a tradition.

File: plutus-core/plutus-core/src/PlutusCore/Error.hs (hunk 1 above)
Comment (zliu41) [reply plutus-core/plutus-core/src/PlutusCore/Error.hs]: Ok, going with the safer option :smile:

**Comment by michaelpj** [comment]
Once we get to 9.2, I'd like to try defining such things with record syntax and `NoFieldSelectors`. That way you can construct the arms of the sum using record syntax (helping with the field confusion problem), but without generating the partial field selector.

**Review by michaelpj — APPROVED** [review]

**Pull request merged**